checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -1,176 +1,878 @@
+   * parameter of the current method are resource aliases.
+   *
+   * @param obligations the obligations associated with the current block
+   * @param argument the argument
+   * @param paramElt the parameter
+   * @return true if {@code paramElt} is a resource alias of {@code arg}
+   */
+  private boolean isParamAndArgAliased(
+      Set<Obligation> obligations, Node argument, VariableElement paramElt) {
+    Set<ResourceAlias> argAliases = getResourceAliasOfArgument(obligations, argument);
+    for (ResourceAlias argAlias : argAliases) {
+      Element argAliasElt = argAlias.reference.getElement();
+      if (argAliasElt.equals(paramElt)) {
+        return true;
       }
     }
+    return false;
   }
 
   /**
-   * Returns the non-exceptional successors of the current block.
+   * Analyzes the called-methods set of each argument passed into the method call to compute
+   * the @Owning annotation for the field or parameter used as an argument in the call.
    *
-   * @param cur the current block
-   * @return the successors of this current block
+   * @param obligations set of obligations associated with the current block
+   * @param invocation a method invocation node to check
    */
-  private List<Block> getNormalSuccessors(Block cur) {
-    List<Block> successorBlock = new ArrayList<>();
+  private void analyzeCalledMethodsOfInvocationArgs(
+      Set<Obligation> obligations, MethodInvocationNode invocation) {
+    List<? extends VariableTree> paramsOfCurrentMethod = methodTree.getParameters();
 
-    if (cur.getType() == Block.BlockType.CONDITIONAL_BLOCK) {
+    List<? extends VariableElement> paramsOfInvocation = mcca.getParametersOfInvocation(invocation);
+    if (paramsOfInvocation.isEmpty()) {
+      return;
+    }
 
-      ConditionalBlock ccur = (ConditionalBlock) cur;
+    for (Node argument : mcca.getArgumentsOfInvocation(invocation)) {
+      Node arg = NodeUtils.removeCasts(argument);
+
+      // In the CFG, explicit passing of multiple arguments in the varargs position is represented
+      // via an ArrayCreationNode. In this case, it checks the called methods set of each argument
+      // passed in this position.
+      if (arg instanceof ArrayCreationNode) {
+        ArrayCreationNode varArgsNode = (ArrayCreationNode) arg;
+        analyzeCalledMethodsSetForVarArgsToComputeOwningParams(
+            obligations, paramsOfCurrentMethod, invocation, varArgsNode, arg);
+      } else {
+        Element varArgElt = TreeUtils.elementFromTree(arg.getTree());
+        if (varArgElt != null && varArgElt.getKind().isField()) {
+          inferOwningField(arg, invocation);
+          continue;
+        }
+        analyzeCalledMethodsSetForArgAliasesToInferOwningParams(
+            obligations, paramsOfCurrentMethod, invocation, arg);
+      }
+    }
+  }
+
+  /**
+   * Analyze each node passed in the varargs argument position. This method checks the
```

Comment:
This description starts with implementation details that a client does not need to know (and should not be burdened with, as the very first part of the documentation).  The documentation should focus on the method's effect.  A better method name could be "computeOwningParamsForVarArgs".